### PR TITLE
Blog post on Kotosiro sharing server

### PR DESCRIPTION
**Pull request by ognis1205**
@dennyglee @MrPowers @JHibbard 
I have finished writing a blog post, would you mind if I ask some of you to review this PR?
Thanks.

### CODE REVIEW

File: src/blog/2023-04-20-open-source-selfhosted-delta-sharing-server/index.mdx [hunk 1]
```diff
@@ -0,0 +1,572 @@
...
+
+![](image1.png)
+
+## Delta Table Structure
+
+You have [historical data on avocado prices and sales volume in multiple US markets](https://www.kaggle.com/datasets/neuromusic/avocado-prices) stored in your Delta table on AWS S3. Your colleague has come to your desk and asked if they could use the data for further data analytics. The structure of the table is as follows:
+
+```
+avocado-table
+├── .part-00000-04d10a18-acde-4d66-bb3b-39f5d0feb689-c000.snappy.parquet.crc
```
Comment (MrPowers) [inline src/blog/2023-04-20-open-source-selfhosted-delta-sharing-server/index.mdx:20]: Can we eliminate the .crc files from the presentation?

File: src/blog/2023-04-20-open-source-selfhosted-delta-sharing-server/index.mdx [hunk 2]
```diff
@@ -0,0 +1,572 @@
...
+│   └── 00000000000000000003.json
+├── part-00000-04d10a18-acde-4d66-bb3b-39f5d0feb689-c000.snappy.parquet
+├── part-00000-c5135c42-2c15-4da5-8cd6-f0fc527dff9c-c000.snappy.parquet
+├── part-00000-c6c1e092-bef3-41a0-8a05-826a33ecff6f-c000.snappy.parquet
+└── part-00000-d7afaec2-4373-4865-ab48-e9f60495b41e-c000.snappy.parquet
+
+2 directories, 16 files
+```
+
+The table is partitioned using the `year` column, and each partition is appended sequentially. Therefore, the table has [four different versions](https://docs.databricks.com/delta/history.html) in chronological order.
```
Comment (MrPowers) [inline src/blog/2023-04-20-open-source-selfhosted-delta-sharing-server/index.mdx:41]: From that file listing, it actually doesn't look like this data is being partitioned on the `year` column.

File: src/blog/2023-04-20-open-source-selfhosted-delta-sharing-server/index.mdx [hunk 3]
```diff
@@ -0,0 +1,572 @@
...
+        <td>organic</td>
+        <td>2017</td>
+        <td>WestTexNewMexico</td>
+      </tr>
+    </tbody>
+  </table>
+  <p>11338 rows × 14 columns</p>
+</div>
+
+[^1]: As of April 20, 2023, this partition filters API is not public, so these code snippets are based on my local patch. I plan to create a pull request in the near future to add these partition filters APIs to the public release.
```
Comment (MrPowers) [inline src/blog/2023-04-20-open-source-selfhosted-delta-sharing-server/index.mdx:561]: Let's just remember to update this whenever you update the code!

File: src/blog/2023-04-20-open-source-selfhosted-delta-sharing-server/index.mdx (hunk 1 above)
Comment (ognis1205) [reply src/blog/2023-04-20-open-source-selfhosted-delta-sharing-server/index.mdx:20]: Sure. I will eliminate them.

File: src/blog/2023-04-20-open-source-selfhosted-delta-sharing-server/index.mdx (hunk 2 above)
Comment (ognis1205) [reply src/blog/2023-04-20-open-source-selfhosted-delta-sharing-server/index.mdx:41]: My bad. You are correct. This is not partitioned (but just appended, each files has the same years). I will delete the sentence.

File: src/blog/2023-04-20-open-source-selfhosted-delta-sharing-server/index.mdx (hunk 3 above)
Comment (ognis1205) [reply src/blog/2023-04-20-open-source-selfhosted-delta-sharing-server/index.mdx:561]: Sure!

**Comment by ognis1205** [comment]
Thanks for the review, @MrPowers !
I fixed them. Would you mind if I ask you review them again?

**Comment by nkarpov** [comment]
LGTM

**Comment by JHibbard** [comment]
LGTM! It seems like the preview generation process isn't working though? I reran the checks but it looks like it hangs after the lint check. @MrPowers, any idea why?

**Comment by dennyglee** [comment]
I kicked off the build but it looks like there are some errors - could you please review?  thanks!

**Comment by ognis1205** [comment]
I will check it out.

**Comment by ognis1205** [comment]
I think the errors were caused by invalid properties passed as JSX properties. I have removed all of them. If the errors still persist, I will replace the corresponding JSX with images. Can you please run the checks again? Thanks.

**Comment by ognis1205** [comment]
I have reviewed the preview and noticed that the table size and footer note appear to be incorrect.
I think I should fix these issues.

**Comment by ognis1205** [comment]
I have addressed all the issues that I noticed. Could you please review it again? Thank you.

**Comment by MrPowers** [comment]
@ognis1205 - your blog post is live now: https://delta.io/blog/2023-04-24-open-source-selfhosted-delta-sharing-server/

Thank you for making this valuable contribution to the community.

**Comment by ognis1205** [comment]
Thanks for your co-operation! @MrPowers @nkarpov @JHibbard @dennyglee

**Pull request merged**